File: llm/lib/decisions.mjs
// Read-only decision-log access for LLM few-shot context.
import fs from 'node:fs';
import path from 'node:path';

export function decisionsDir(kbRoot) {
  return path.join(kbRoot, '.kb', 'govern', 'decisions');
}

export function listDecisions(kbRoot) {
  const dir = decisionsDir(kbRoot);
  if (!fs.existsSync(dir)) return [];
  return fs.readdirSync(dir)
    .filter((f) => f.endsWith('.json'))
    .map((f) => path.join(dir, f));
}

export function readDecision(kbRoot, id) {
  const p = path.join(decisionsDir(kbRoot), `${id}.json`);
  if (!fs.existsSync(p)) return null;
  return JSON.parse(fs.readFileSync(p, 'utf8'));
}

export function decisionsByType(kbRoot, decisionType) {
  const out = [];
  for (const p of listDecisions(kbRoot)) {
    try {
      const rec = JSON.parse(fs.readFileSync(p, 'utf8'));
      if (rec.decision_type === decisionType) out.push(rec);
    } catch { /* skip malformed */ }
  }
  return out.sort((a, b) => (a.ts || '').localeCompare(b.ts || ''));
}

File: agent/weft_agent/decisions.py
"""Read-only decision-log access for LLM few-shot context (port of llm/lib/decisions.mjs)."""
import json
from pathlib import Path


def decisions_dir(kb_root: Path) -> Path:
    return kb_root / ".kb" / "govern" / "decisions"


def list_decisions(kb_root: Path) -> list[Path]:
    d = decisions_dir(kb_root)
    if not d.is_dir():
        return []
    return sorted(d.glob("*.json"))


def read_decision(kb_root: Path, decision_id: str) -> dict | None:
    p = decisions_dir(kb_root) / f"{decision_id}.json"
    if not p.exists():
        return None
    return json.loads(p.read_text(encoding="utf-8"))


def decisions_by_type(kb_root: Path, decision_type: str) -> list[dict]:
    out = []
    for p in list_decisions(kb_root):
        try:
            rec = json.loads(p.read_text(encoding="utf-8"))
        except (json.JSONDecodeError, OSError):
            continue  # skip malformed
        if rec.get("decision_type") == decision_type:
            out.append(rec)
    out.sort(key=lambda r: r.get("ts") or "")
    return out
